server/actions/document.action.js

```diff
@@ -0,0 +1,192 @@
+            });
+          }
+        });
+    });
+}
+
+/**
+ * Fetches all documents
+ * that the user has permission to view
+ * @exports getAll
+ * @param {any} req
+ * @param {any} res
+ * @returns {any} res
+ */
+export function getAll(req, res) {
+  if (req.decoded.role === 'admin') {
+    Document.findAll().then((documents) => {
+      res.status(200).json(documents);
+    });
+  } else {
+    Document.findAll({
+      where: {
+        $or: [{
+          userId: req.decoded.id
+        }, {
+          access: 'public'
+        }]
+      }
+    }).then((documents) => {
+      res.status(200).json(documents);
+    });
+  }
+}
+
+/**
+ * Fetches all role access documents
+ * that the user has permission to view
+ * @exports getAll
+ * @param {any} req
+ * @param {any} res
+ * @returns {any} res
+ */
+export function getAllForRole(req, res) {
+  Role.findOne({
+    where: {
+      title: req.decoded.role
+    }
+  }).then((role) => {
+    Document.findAll({
+      where: {
+        $and: [{
+          access: 'role'
+        }, {
+          roleId: role.id
+        }]
+      }
+    }).then((documents) => {
+      res.status(200).json(documents);
+    });
+  });
+}
+
+/**
+ * Fetches all Documents
+ * created by a user
+ * @exports getAll
+ * @param {any} req
+ * @param {any} res
+ * @returns {any} res
+ */
+export function getAllByUser(req, res) {
+  Document.findAll({ where: { userId: req.decoded.id } })
+    .then((documents) => {
+      res.status(400).json(documents);
+    });
+}
+
+/**
+ * Creates a Document
+ * @exports create
+ * @param {any} req
+ * @param {any} res
+ * @returns {any} res
+ */
+export function create(req, res) {
+  Role.findOne({ where: { title: req.decoded.role } })
+    .then((role) => {
+      const newDocument = {};
+      Object.assign(newDocument, req.body, { userId: req.decoded.id, roleId: role.id });
+      Document.create(newDocument)
+        .then((document) => {
+          res.status(201).json({
+            message: 'Document created successfully',
+            document
+          });
+        })
+        .catch((err) => {
+          res.status(422).json({
+            message: (() => {
+              const message = [];
+              err.errors.map((error) => { // eslint-disable-line array-callback-return
+                message.push(error.message);
+              });
+              return message;
+            })()
+          });
+        });
+    });
+}
+
+/**
+ * Edits a Document
+ * @exports edit
+ * @param {any} req
+ * @param {any} res
+ *@returns {any} res
+ */
+export function edit(req, res) {
+  Document.update(req.body, {
+    where: {
+      $and: [{
+        id: req.params.id
+      }, {
+        userId: req.decoded.id
+      }]
+    }
+  }).then((numberOfRows) => {
+    if (numberOfRows > 0) {
+      return res.status(200).json({
+        message: 'Document edited successfully'
+      });
+    } else { // eslint-disable-line no-else-return
```

Comment:
@andela-badebiyi. That's true. You're right. it returns an empty array. What I'm after is the eradication of the if else statement. it shows the function is doing more than one thing.. 
This function can be re-written as this
```javascript
if (numberOfRows > 0) {
     return res.status(200).json({
       message: 'Document edited successfully'
 });
 return res.status(404).json({
      message: 'Document not found'
 });
```
Or more explicitly since 404 is an error message it can be rewritten like this
```javascript
if (numberOfRows == 0) throw new Error('Document Not found');
return res.status(200).json({
       message: 'Document edited successfully'
 });
).catch(err => res.status(404).message(err))
```